### PR TITLE
Apply autocomplete redesign from #740, format the displayed URLs in

### PR DESCRIPTION
**Pull request by jsantell**
autocomplete, and center the results. r=vporof,phlsa

### CODE REVIEW

**Comment by jsantell** [comment]
This handles many of the changes in #740; filed a follow up for the magnifying glass. This pretty formats the URLs, handles overflow, centers the content, updates colors etc.

Things not addressed in mockup in this PR:
* True centering for the autocomplete content to be aligned perfectly with the location bar
* Fading out the title/url when overflows (ellipsis used instead currently)
* Have a separate hover state for the rows

**Comment by jsantell** [comment]
screen shot 2016-07-11 at 1 32 43 pm

**Comment by victorporof** [comment]
nice!! reviewing.

**Comment by phlsa** [comment]
Looks good!
I just realized that there was an error with the hover state in my original mocks (the hover background faded out with the text). Here's the corrected version:

tofino-labels

tofino

**Comment by jsantell** [comment]
screen shot 2016-07-12 at 9 36 44 am

Ok, gradients added, and fixed the part where I accidentally kept the code that kept the autocomplete window to stay open for style inspection

File: app/ui/browser/views/navbar/location-completion-row.jsx [hunk 1]
```diff
@@ -13,29 +13,56 @@ specific language governing permissions and limitations under the License.
 import React, { PropTypes } from 'react';
 
 import Style from '../../../shared/style';
+import { prettyUrl } from '../../../../shared/url';
+
+const GRADIENT_MASK = `-webkit-linear-gradient(left,
```
Comment (victorporof) [inline app/ui/browser/views/navbar/location-completion-row.jsx]: Can we avoid webkit-prefixed gradients?

File: app/ui/browser/views/navbar/location-completion-row.jsx [hunk 2]
```diff
@@ -13,29 +13,56 @@ specific language governing permissions and limitations under the License.
...
+                      rgba(0, 0, 0, 1) 0%,
+                      rgba(0, 0, 0, 1) 90%,
+                      rgba(0, 0, 0, 0) 95%)`;
 
 const ROW_STYLE = Style.registerStyle({
-  // We're nesting multi-line items inside a flexbox, so
-  // we need to mark these children as columnar.
-  flexDirection: 'column',
+  display: 'flex',
```
Comment (victorporof) [inline app/ui/browser/views/navbar/location-completion-row.jsx]: display: flex is redundant here, all divs are flexed.

File: app/ui/browser/views/navbar/location-completion-row.jsx [hunk 3]
```diff
@@ -13,29 +13,56 @@ specific language governing permissions and limitations under the License.
...
+  padding: '5px',
+});
+
+const TITLE_STYLE = Style.registerStyle({
+  overflow: 'hidden',
+  whiteSpace: 'nowrap',
+  paddingRight: '10px',
+  flex: 6,
+  '-webkit-mask-image': GRADIENT_MASK,
```
Comment (victorporof) [inline app/ui/browser/views/navbar/location-completion-row.jsx]: Ugh. Nothing standard we could use?

File: app/ui/browser/views/navbar/location-completion-row.jsx [hunk 4]
```diff
@@ -13,29 +13,56 @@ specific language governing permissions and limitations under the License.
...
+});
+
+const URL_STYLE = Style.registerStyle({
+  flex: 4,
+  color: 'var(--theme-body-color)',
+  fontSize: '11px',
```
Comment (victorporof) [inline app/ui/browser/views/navbar/location-completion-row.jsx]: Let's avoid fixed font sizes across the app, use percentages.

File: app/ui/browser/views/navbar/location-completion-row.jsx [hunk 5]
```diff
@@ -13,29 +13,56 @@ specific language governing permissions and limitations under the License.
...
 const FOCUSED_RESULT_STYLE = Style.registerStyle({
-  background: 'var(--theme-selection-color)',
-  color: 'var(--theme-content-selected-color)',
+  backgroundColor: 'var(--theme-selection-color)',
 });
 
+/*
 const COMPLETION_TEXT_STYLE = Style.registerStyle({
-  color: 'rgba(100, 100, 100, 1)',
+  color: '#333',
```
Comment (victorporof) [inline app/ui/browser/views/navbar/location-completion-row.jsx]: Why do we keep this around?

File: app/ui/browser/views/navbar/location.jsx [hunk 6]
```diff
@@ -54,8 +54,9 @@ const LOCATION_BAR_AUTOCOMPLETE_STYLE = Style.registerStyle({
   flexDirection: 'column',
   position: 'absolute',
   background: '#fcfcfc',
-  color: 'var(--theme-body-color)',
-  padding: 10,
+  color: '#000',
```
Comment (victorporof) [inline app/ui/browser/views/navbar/location.jsx]: Should have css vars for text colors.

File: app/ui/browser/views/navbar/location.jsx [hunk 7]
```diff
@@ -54,8 +54,9 @@ const LOCATION_BAR_AUTOCOMPLETE_STYLE = Style.registerStyle({
...
+  fontSize: '14px',
```
Comment (victorporof) [inline app/ui/browser/views/navbar/location.jsx]: No fixed font sizes please.

File: app/ui/shared/css/theme.css [hunk 8]
```diff
@@ -2,7 +2,7 @@
   --theme-body-color: #2d9cdb;
   --theme-content-color: #fff;
   --theme-content-selected-color: #fff;
-  --theme-selection-color: rgba(47,128,237,0.5);
+  --theme-selection-color: #B3DBF2;
```
Comment (victorporof) [inline app/ui/shared/css/theme.css]: Uber nit: lowercase hex.

File: app/ui/browser/views/navbar/location-completion-row.jsx (hunk 3 above)
Comment (jsantell) [inline app/ui/browser/views/navbar/location-completion-row.jsx]: Looks like only the prefixed version is supported, and same with the linear-gradient, as use by the mask-image :\

**Pull request merged**